src/pages/timer-settings/timer-settings.ts

```diff
@@ -0,0 +1,57 @@
+import { TimerStorageProvider } from './../../providers/timer-storage/timer-storage';
+import { TimerSettingStorageProvider } from './../../providers/timer-setting-storage/timer-setting-storage';
+import { TimerSetting } from './../../classes/timer-settings';
+import { SingleTimer } from './../../classes/single-timer';
+import { Component } from '@angular/core';
+import { IonicPage, NavController, NavParams } from 'ionic-angular';
+
+/**
+ * Generated class for the TimerSettingsPage page.
+ *
+ * See https://ionicframework.com/docs/components/#navigation for more info on
+ * Ionic pages and navigation.
+ */
+
+@IonicPage()
+@Component({
+  selector: 'page-timer-settings',
+  templateUrl: 'timer-settings.html',
+})
+export class TimerSettingsPage {
+
+  addNew: boolean = true;
+  editTimer: SingleTimer;
+  timerSettings: TimerSetting;
+
+  constructor(
+    public navCtrl: NavController, 
+    public navParams: NavParams,
+    private settingStorage: TimerSettingStorageProvider,
+    private timerStorage: TimerStorageProvider
+  ) {
+    this.editTimer = this.navParams.get('timer');
+    if ( this.editTimer ) {
+      this.addNew = false;
+      this.editTimer = JSON.parse(JSON.stringify(this.editTimer))
```

Comment:
You have an object. Parse that to a string and parse the string to an object again?

Reply:
Yes. It's one of the easiest way to deep clone an object